omp/CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-find_package(OpenMP 3.0 REQUIRED)
+find_package(OpenMP 4.0 REQUIRED)
```

Comment:
Isn't this quite a significant change? Is it considered interface breaking when we move to a newer OpenMP version?

Reply:
not really in terms of compiler support: All of our compilers support it AFAIK, except for MSVC, but they are behind anyways

Reply:
Okay, almost, I didn't know we still test for [Clang 7.0](https://gitlab.com/ginkgo-project/ginkgo-public-ci/-/jobs/2311759914). I can also do 3.1, but that means that we can't require sequential consistency semantics from the atomics, which I would be more comfortable with than just using `#pragma omp atomic *`